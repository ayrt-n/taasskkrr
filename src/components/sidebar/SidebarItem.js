import React from 'react';
import { Link } from 'react-router-dom';

function SidebarItem({ icon, title, action, subItem }) {
  const iconSrc = require(`../../assets/icons/${icon}`)

  return (
    <Link to={action}>
      <div className={`SidebarItem ${subItem ? "SubItem" : ""}`}>
        <img src={iconSrc} alt="" className="SidebarItem-icon" />
        <div className="SidebarItem-title">{title}</div>
      </div>
    </Link>
  );
}

export default SidebarItem;
