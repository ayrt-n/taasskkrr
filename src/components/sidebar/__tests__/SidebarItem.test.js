import React from 'react';
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom'
import SidebarItem from '../SidebarItem';

describe('SidebarItem component', () => {
  it('correctly renders sidebar item', () => {
    const { container } = render(
      <SidebarItem icon='icon.jpg' title='Sidebar Item' action='/sidebar-action' />,
      {wrapper: MemoryRouter}
    );

    expect(container).toMatchSnapshot();
  });

  it('correctly renders sidebar sub-item', () => {
    const { container } = render(
      <SidebarItem icon='icon.jpg' title='Sidebar Sub-Item' action='/sidebar-action' subItem={true} />,
      {wrapper: MemoryRouter}
    );

    expect(container).toMatchSnapshot();
  });
});