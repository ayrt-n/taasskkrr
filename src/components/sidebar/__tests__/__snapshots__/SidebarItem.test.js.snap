// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem component correctly renders sidebar item 1`] = `
<div>
  <a
    href="/sidebar-action"
  >
    <div
      class="SidebarItem "
    >
      <img
        alt=""
        class="SidebarItem-icon"
        src="test-file-stub"
      />
      <div
        class="SidebarItem-title"
      >
        Sidebar Item
      </div>
    </div>
  </a>
</div>
`;

exports[`SidebarItem component correctly renders sidebar sub-item 1`] = `
<div>
  <a
    href="/sidebar-action"
  >
    <div
      class="SidebarItem SubItem"
    >
      <img
        alt=""
        class="SidebarItem-icon"
        src="test-file-stub"
      />
      <div
        class="SidebarItem-title"
      >
        Sidebar Sub-Item
      </div>
    </div>
  </a>
</div>
`;
